src/CMakeLists.txt

```diff
@@ -166,6 +166,11 @@ set(MEMORYCHECK_COMMAND_OPTIONS "--track-origins=yes --leak-check=full --trace-c
 option(BUILD_TESTING "Build the testing tree." OFF)
 include(CTest)
 
+option(WITH_PROCESS "Enable process management feature, used for testing." ON)
+if (WITH_PROCESS)
+  add_definitions(-DPROCESS_MANAGEMENT_ENABLED)
```

Comment:
Please never, ever, just call *add_definitions*. It does not do what you want it to. When adding definitions for a target, use *target_compile_definitions instead*. However, please remove it entirely for now. I'll provide a decent example for making things optional in my FreeRTOS work.